src/main/java/software/amazon/awssdk/crt/http/HttpClientConnection.java

```diff
@@ -23,6 +25,42 @@
  * This class is not thread safe and should not be called from different threads.
  */
 public class HttpClientConnection extends CrtResource {
+    /**
+     * HTTP protocol version.
+     */
+    public enum ProtocolVersion {
+        HTTP_1_0(1),
+        HTTP_1_1(2),
+        HTTP_2(3);
+
+        private int protocolVersion;
+        private static Map<Integer, ProtocolVersion> enumMapping = buildEnumMapping();
+
+        ProtocolVersion(int value) {
+            protocolVersion = value;
+        }
+
+        public static ProtocolVersion getEnumValueFromInteger(int value) {
```

Comment:
nit: not sure how closely you try to follow the Java SDK conventions but this is normally named `fromValue`